radix-transactions/src/manifest/static_resource_movements/visitor.rs

```diff
@@ -129,53 +129,81 @@ impl StaticResourceMovementsVisitor {
         &self,
         invocation_kind: InvocationKind,
         args: &ManifestValue,
-    ) -> Result<Option<(TypedNativeInvocation, InvocationReceiver)>, StaticResourceMovementsError>
-    {
+    ) -> Result<
+        Option<(TypedManifestNativeInvocation, InvocationReceiver)>,
+        StaticResourceMovementsError,
+    > {
         // Creating a typed native invocation to use in interpreting the invocation.
-        Ok(match invocation_kind {
-            InvocationKind::Method {
-                address: DynamicGlobalAddress::Static(global_address),
-                module_id: ModuleId::Main,
-                method,
-            } => TypedNativeInvocation::from_main_module_method_invocation(
-                global_address,
-                method,
-                args,
-            )?
-            .map(|value| (value, InvocationReceiver::GlobalMethod(*global_address))),
+        match invocation_kind {
+            InvocationKind::DirectMethod { address, method } => {
+                let resolved_dynamic_address = ResolvedDynamicAddress::StaticAddress(*address);
+                let Some(typed_invocation) = TypedManifestNativeInvocation::from_method_invocation(
```

Comment:
Arguably this should be `from_direct_method_invocation` - because I believe direct methods are considered separately. (And we'd have a separate list of direct methods for each blueprint).

e.g. To be concrete, you can't direct call take on a fungible vault, but it looks like this code will allow it - which could lead to confusion.

Reply:
I think you're right that this would be allowed. We could add a todo comment here about this and that it might be a good idea for direct methods to be separated into their own enum in the future. For the time being, all of the direct methods that we have output unknown resources anyway so we don't quite gain a lot from doing this.

Reply:
> Ahh - so are you saying we don't support recall right now?

We do support it, but we can'te tell anything meaningful about a recall since we're missing the resource address.